pydis_core/_bot.py

```diff
@@ -116,18 +117,31 @@ def _connect_statsd(
                 attempt + 1
             )
 
-    async def load_extensions(self, module: types.ModuleType) -> None:
-        """
-        Load all the extensions within the given module and save them to ``self.all_extensions``.
-
-        This should be ran in a task on the event loop to avoid deadlocks caused by ``wait_for`` calls.
-        """
+    async def _load_extensions(self, module: types.ModuleType) -> None:
+        """Load all the extensions within the given module and save them to ``self.all_extensions``."""
         await self.wait_until_guild_available()
         self.all_extensions = walk_extensions(module)
 
         for extension in self.all_extensions:
             scheduling.create_task(self.load_extension(extension))
 
+    async def _sync_app_commands(self) -> None:
+        """Sync global & guild specific application commands after extensions are loaded."""
+        await self._extension_loading_task
+        await self.tree.sync()
+        await self.tree.sync(guild=discord.Object(self.guild_id))
+
+    async def load_extensions(self, module: types.ModuleType, sync_app_commands: bool = True) -> None:
+        """
+        Load all the extensions within the given ``module`` and save them to ``self.all_extensions``.
+
+        Args:
+            sync_app_commands: Whether to sync app commands after all extensions are loaded.
+        """
+        self._extension_loading_task = scheduling.create_task(self._load_extensions(module))
+        if sync_app_commands:
```

Comment:
I'm just curious here, are you making this possibly optional for people who wouldn't want to sync their commands ? 
If yes, is there particular reason behind this ? (Perf, etc.)

Reply:
Yea, we might not always want slash commands loaded in dev if we end up with a lot of them, as there is a rate limit on how many new commands can be registered each day

Reply:
Understandable, thanks !